.github/workflows/docker.yml

```diff
@@ -23,9 +27,34 @@ jobs:
           username: NordicBuilder
           password: ${{ secrets.NCS_GITHUB_PKG_WRITE_TOKEN }}
 
+      - name: Check if Docker image exists
+        id: check-image
+        run: |
+          TOOLCHAIN_ID=$(./scripts/print_toolchain_checksum.sh)
+          echo "TOOLCHAIN_ID=$TOOLCHAIN_ID" >> $GITHUB_OUTPUT
+          IMAGE_NAME=ghcr.io/${{ github.repository }}-toolchain:$TOOLCHAIN_ID
```

Comment:
nit: should be possible to write image name to GITHUB_ENV here once and delete multiple lines setting the same image_name again.